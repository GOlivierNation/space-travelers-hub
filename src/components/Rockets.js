import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { reserveRocket, cancelReseve, fetchData } from '../redux/rockets/rockets';

const Rockets = () => {
  const displayRockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchData());
  }, []);

  const handleReserve = (id) => {
    dispatch(reserveRocket(id));
  };

  const handleCancel = (id) => {
    dispatch(cancelReseve(id));
  };

  return (
    <div className="rocket-container">
      {displayRockets.map((rocket, index) => (
        <div key={[index]} className="rocket">
          <img className="rocket-img" src={rocket.image} alt={rocket.name} />
          <div className="rocket-info">
            <h2>{rocket.name}</h2>
            <p>
              {(rocket.reserved) ? <span>Reserved</span> : '' }
              {rocket.description}
            </p>
            {(rocket.reserved) ? <button type="button" className="canceled" onClick={() => handleCancel(rocket.id)}>Cancel Reservation</button>
              : <button type="button" className="booked" onClick={() => handleReserve(rocket.id)}>Reserve Rocket</button> }
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
