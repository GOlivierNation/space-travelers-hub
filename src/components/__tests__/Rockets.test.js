import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Rockets from '../Rockets';

describe('Tests for the Rockets page', () => {

  it('Should render the Rockets page', () => {
    const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});