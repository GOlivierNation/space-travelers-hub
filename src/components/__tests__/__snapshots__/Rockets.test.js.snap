// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Rockets page Should render the Rockets page 1`] = `
<div
  className="rocket-container"
/>
`;
